### PR TITLE
feat(wallet-dasboard): style vesting schedule

### LINKED ISSUE #4264: Style vesting schedule
Depends on https://github.com/iotaledger/iota/issues/4259
Style vesting schedule
Image

### PR DESCRIPTION
**Pull request by evavirseda**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/4264

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 1]
```diff
@@ -92,6 +94,9 @@ function VestingDashboardPage(): JSX.Element {
         false,
     );
 
+    const vestingPortfolio =
+        nextPayout && buildSupplyIncreaseVestingSchedule(nextPayout, Number(currentEpochMs));
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: We need the lastPayout to build the schedule

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 2]
```diff
@@ -252,17 +260,19 @@ function VestingDashboardPage(): JSX.Element {
...
-                                /*Open schedule dialog*/
-                            }}
+                            onClick={openReceiveTokenPopup}
                             title="See All"
                             buttonType={ButtonType.Secondary}
-                            buttonDisabled={
-                                !vestingSchedule.availableStaking ||
-                                vestingSchedule.availableStaking === 0
-                            }
+                            buttonDisabled={!vestingSchedule}
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
                            buttonDisabled={!vestingPortfolio}
```

File: apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    amount: number;
+    expirationTimestampMs: number;
+}
+
+export function VestingScheduleBox({
+    amount,
+    expirationTimestampMs,
+}: VestingScheduleBoxProps): React.JSX.Element {
+    const [formattedAmountVested, amountVestedSymbol] = useFormatCoin(amount, IOTA_TYPE_ARG);
+    const isLocked = expirationTimestampMs > Date.now();
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx]:
```suggestion
    const isLocked = expirationTimestampMs > currentEpochTimestamp;
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx (hunk 3 above)
Comment (cpl121) [reply apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx]: Im not sure about this, let me ask

**Review by brancoder — APPROVED** [review]
LGTM 👍

File: apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx (hunk 3 above)
Comment (cpl121) [reply apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx]: okay, all clear now
I will create new comments with the changes 🙏🏼

File: apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    amount,
+    expirationTimestampMs,
+}: VestingScheduleBoxProps): React.JSX.Element {
+    const [formattedAmountVested, amountVestedSymbol] = useFormatCoin(amount, IOTA_TYPE_ARG);
+    const { data: currentEpochMs } = useGetCurrentEpochStartTimestamp();
+
+    const isLocked = expirationTimestampMs > Number(currentEpochMs);
+    return (
+        <DisplayStats
+            label={new Date(expirationTimestampMs).toLocaleString()}
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Dialogs/vesting/VestingScheduleBox.tsx]: we will only show the date until we know what time to show
```suggestion
            label={new Date(expirationTimestampMs).toLocaleDateString()}
```
https://github.com/iotaledger/iota/issues/4369

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by cpl121 — CHANGES_REQUESTED** [review]
build error 🙏🏼

**Review by brancoder — CHANGES_REQUESTED** [review]
Lint 🙏

File: apps/wallet-dashboard/lib/utils/vesting/vesting.spec.ts [hunk 5]
```diff
@@ -20,6 +20,8 @@ import {
     getSupplyIncreaseVestingUserType,
 } from './vesting';
 
+const MOCKED_CURRENT_EPOCH_TIMESTAMP = 1733340503570; // Wednesday, 04 December 2024 19:28:23
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.spec.ts]: It is better that this mocked constant is dynamic
```suggestion
const MOCKED_CURRENT_EPOCH_TIMESTAMP = Date.now() + (MILLISECONDS_PER_HOUR * 6); // 6 hours later
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by cpl121 — APPROVED** [review]
NJ Eva 👏🏼

**Review by brancoder — APPROVED** [review]
LGTM 🎊

**Pull request merged**